fetchai/ledger/parser/etch_parser.py

```diff
@@ -179,22 +182,40 @@ class UnparsableAddress(Exception):
     pass
 
 
+class NoParsedEtchSource(Exception):
+    pass
+
+
 class EtchParser:
     def __init__(self, etch_code=None):
         # Load grammar
         self.grammar = resource_string(__name__, 'etch.grammar').decode('ascii')
         self.parser = Lark(self.grammar, propagate_positions=True)
 
-        self.parsed_tree = None
+        self._parsed_tree = None
         self.etch_code = None
         if etch_code:
             self.parse(etch_code)
 
+    @property
+    def parsed_tree(self):
+        if self._parsed_tree:
+            return self._parsed_tree
+        else:
+            raise NoParsedEtchSource()
+
     def parse(self, etch_code):
         """Parses the input code and stores the parsed tree"""
         assert isinstance(etch_code, str), "Expecting string"
         self.etch_code = etch_code
-        self.parsed_tree = self.parser.parse(etch_code)
+        try:
+            self._parsed_tree = self.parser.parse(etch_code)
+        except (LarkError, GrammarError, ParseError, LexError, UnexpectedInput,
+                UnexpectedCharacters, UnexpectedToken, VisitError) as e:
+            logging.warning("Etch parsing failed, shard masks will be set to wildcard")
```

Comment:
I wouldn't mention a parse failure here, as it may not be the user's fault (i.e. their etch code is correct, but the parser is throwing a wobbly; we ignore this, because this isn't our reference parser). This phrasing may spook users and send them on a wild goose chase. looking for errors in their code.

I'd just say something like "Unable to autodetect shard mask; defaulting to wildcard" here.

Reply:
I agree with this in principle, but in practice, the parsers blind spots are fewer and fewer, and this error will be encountered far more often for actually invalid code, and in that case it's potentially a lot of work to discover what the issue is.